deploy/kubernetes/webhook-example/README.md

```diff
@@ -0,0 +1,87 @@
+# Validating Webhook
+
+The snapshot validating webhook is an HTTP callback which responds to [admission requests](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/). It is part of a larger [plan](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/177-volume-snapshot/tighten-validation-webhook-crd.md) to tighten validation for volume snapshot objects. This webhook introduces the [ratcheting validation](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/177-volume-snapshot/tighten-validation-webhook-crd.md#backwards-compatibility) mechanism targeting the tighter validation.
+
+> :warning: **WARNING**: Choosing not to install the webhook server and participate in the phased release process can cause future problems when upgrading from `v1beta1` to `v1` volumesnapshot API if there are currently persisted objects which fail the new stricter validation. Potential impacts include being unable to delete invalid snapshot objects.
+
+## Prerequisites
+
+The following are prerequisites to use this validating webhook:
+
+- K8s version 1.17+ (v1.9+ to use `admissionregistration.k8s.io/v1beta1`, v1.16+ to use `admissionregistration.k8s.io/v1`, v1.17+ to use  `snapshot.storage.k8s.io/v1beta1`)
+- ValidatingAdmissionWebhook is [enabled](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#prerequisites). (in v1.18+ it will be enabled by default)
+- API `admissionregistration.k8s.io/v1beta1` or `admissionregistration.k8s.io/v1` is enabled.
+
+## How to build the webhook
+
+Build the binary
+
+```bash
+make 
+```
+
+Build the docker image
+
+```bash
+docker build -t gcr.io/your-project-name/webhook:latest -f ./cmd/webhook/Dockerfile .
+```
+
+Push the docker image to the remote registry
+
+```bash
+docker push gcr.io/your-project-name/webhook:latest
+```
```

Comment:
I think the steps to push image to remote registry can be skipped as they are not specific to this project.  Local images work fine if I'm running a local cluster.

Reply:
Removed this section